src/config.rs

```diff
@@ -37,188 +39,55 @@ impl Default for Config {
-        self.use_compression = use_compression;
-    }
-
-    /// Set the number of milliseconds between flushes.
-    pub fn set_flush_every_ms(&mut self, flush_every_ms: Option<u64>) {
-        self.flush_every_ms = flush_every_ms;
-    }
-
-    /// Builder, set number of io buffers
-    pub fn io_bufs(&self, io_bufs: usize) -> Config {
-        let mut ret = self.clone();
-        ret.io_bufs = io_bufs;
-        ret
-    }
-
-    /// Builder, set the max io buffer size
-    pub fn io_buf_size(&self, io_buf_size: usize) -> Config {
-        let mut ret = self.clone();
-        ret.io_buf_size = io_buf_size;
-        ret
-    }
-
-    /// Builder, set the b-link tree node fanout
-    pub fn blink_fanout(&self, blink_fanout: usize) -> Config {
-        let mut ret = self.clone();
-        ret.blink_fanout = blink_fanout;
-        ret
-    }
-
-    /// Builder, set the pagecache consolidation threshold
-    pub fn page_consolidation_threshold(&self, threshold: usize) -> Config {
-        let mut ret = self.clone();
-        ret.page_consolidation_threshold = threshold;
-        ret
-    }
-
-    /// Builder, set the filesystem path
-    pub fn path(&self, path: Option<String>) -> Config {
-        let mut ret = self.clone();
-        ret.path = path;
-        ret
-    }
-
-    /// Builder, set the number of bits used for addressing cache shards.
-    pub fn cache_bits(&self, cache_bits: usize) -> Config {
-        let mut ret = self.clone();
-        ret.cache_bits = cache_bits;
-        ret
-    }
-
-    /// Builder, set the cache capacity in bytes.
-    pub fn cache_capacity(&self, cache_capacity: usize) -> Config {
-        let mut ret = self.clone();
-        ret.cache_capacity = cache_capacity;
-        ret
-    }
+macro_rules! builder {
+    ($(($name:ident, $get:ident, $set:ident, $t:ty, $desc:expr)),*) => {
+        $(
+            #[doc="Get "]
+            #[doc=$desc]
+            pub fn $get(&self) -> $t {
+                self.$name.clone()
+            }
 
-    /// Builder, set whether the system uses the OS pagecache for
-    /// compressed pages.
-    pub fn use_os_cache(&self, use_os_cache: bool) -> Config {
-        let mut ret = self.clone();
-        ret.use_os_cache = use_os_cache;
-        ret
-    }
+            #[doc="Set "]
+            #[doc=$desc]
+            pub fn $set(&mut self, to: $t) {
+                self.$name = to;
+            }
 
-    /// Builder, set whether the system compresses data written to
-    /// secondary storage with the zstd algorithm.
-    pub fn use_compression(&self, use_compression: bool) -> Config {
-        let mut ret = self.clone();
-        ret.use_compression = use_compression;
-        ret
+            #[doc="Builder, set "]
+            #[doc=$desc]
+            pub fn $name(&self, to: $t) -> Config {
+                let mut ret = self.clone();
+                ret.$name = to;
+                ret
+            }
+        )*
```

Comment:
it's also possible to add a custom separator before the `*` if necessary, but it's not here